### ISSUE TITLE
Unhandled JSON.parse errors

### ISSUE THREAD

**Issue by cjroebuck** [jshemas/openGraphScraper#247]
**Describe the bug**
JSON.parse errors are not handled correctly, causing error result to be thrown rather than returned in the Promise.

**To Reproduce**
Try it on the html output of for e.g. https://www.headlightsdepot.com because one of the JSON+LD script tags in there has line breaks which causes JSON.parse to throw.

**Expected behavior**
Expected to be able to use the error result object to handle the error, but had to wrap the call to `await ogs` in try/catch and handle the error result in the catch branch.

**Actual behavior**
The call to `await ogs({html})` throws the ErrorResult

ogs version: 6.8.1

**Comment by jshemas** [comment]
I'm getting `UND_ERR_HEADERS_OVERFLOW`( HeadersOverflowError: Headers Overflow Error) errors from this page. Are these the errors you are seeing? If yes, you will need to open a issue on `undici`.

Example code:

```
const { fetch } = require('undici');

const getAPI = async () => {
  const request = await fetch('https://www.headlightsdepot.com');
  const text = await request.text();
  console.log('text:', text);
};

getAPI();
```

**Comment by jshemas** [comment]
Hello, thanks for the full error. This should be fixed in `open-graph-scraper@6.8.2`.

**Comment by jshemas** [comment]
Closing this for now. Please open a new ticket if you see any other issues.
